wandb/sdk/verify/verify.py

```diff
@@ -167,7 +179,8 @@ def check_run(api: Api) -> None:
             "Read summary values don't match expected value. Check database encoding, or contact W&B for support."
         )
     try:
-        read_file = prev_run.file(filepath).download(replace=True)
+        read_file = retry_fn(partial(prev_run.file, filepath))
+        read_file = read_file.download(replace=True)
     except Exception:
         with wandb.init(
```

Comment:
This structuring of running significant code inside an exception handler should be avoided.
Add a TODO to refactor this.